From 8886904aabd99d633d9a7e6e55fcab6f513e818f Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 26 Aug 2014 12:49:39 -0400
Subject: [tools] cpupower: mperf monitor, Correct use of ! and &

Message-id: <1409057382-26432-29-git-send-email-prarit@redhat.com>
Patchwork-id: 89101
O-Subject: [RHEL7.1 PATCH BZ 1133893 28/31] cpupower: mperf monitor: Correct use of ! and &
Bugzilla: 1133893
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1133893

commit 97fa1c5ca680bdee2c650e3aadf2a839b92f3f0e
Author: Himangi Saraogi <himangi774@gmail.com>
Date:   Tue Jul 29 18:12:18 2014 +0200

    cpupower: mperf monitor: Correct use of ! and &

    In commit ae91d60ba88ef0bdb1b5e9b2363bd52fc45d2af7, a bug was fixed that
    involved converting !x & y to !(x & y).  The code below shows the same
    pattern, and thus should perhaps be fixed in the same way.

    The Coccinelle semantic patch that makes this change is as follows:

    // <smpl>
    @@ expression E1,E2; @@
    (
      !E1 & !E2
    |
    - !E1 & E2
    + !(E1 & E2)
    )
    // </smpl>

    Signed-off-by: Himangi Saraogi <himangi774@gmail.com>
    Signed-off-by: Thomas Renninger <trenn@suse.de>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/cpupower/utils/idle_monitor/mperf_monitor.c b/tools/power/cpupower/utils/idle_monitor/mperf_monitor.c
index 5650ab5..90a8c4f 100644
--- a/tools/power/cpupower/utils/idle_monitor/mperf_monitor.c
+++ b/tools/power/cpupower/utils/idle_monitor/mperf_monitor.c
@@ -237,7 +237,7 @@ static int init_maxfreq_mode(void)
  unsigned long long hwcr;
  unsigned long min;
 
- if (!cpupower_cpu_info.caps & CPUPOWER_CAP_INV_TSC)
+ if (!(cpupower_cpu_info.caps & CPUPOWER_CAP_INV_TSC))
   goto use_sysfs;
 
  if (cpupower_cpu_info.vendor == X86_VENDOR_AMD) {
-- 
1.7.1